src/main/test/com/epam/preprod/sirenko/strategy/impl/AutoGeneratedProductCreationStrategyTest.java

```diff
@@ -3,75 +3,98 @@
+	
+	@InjectMocks
+	AutoGeneratedProductCreationStrategy autoGeneratedProductCreationStrategy;
+	
+	@Mock
+	Random random;
 	
 	@BeforeEach
 	public void SetData() {
-		autoGeneratedProductCreation = new AutoGeneratedProductCreationStrategy();
-		strategy = mock(Strategy.class);
-		random = mock(Random.class);
+		MockitoAnnotations.initMocks(this);
 	}
 	
 	@Test
-	void getInt() {
-		when(random.nextInt(anyInt())).thenReturn(10);
-		//TODO doesn't work
-		assertEquals(10, autoGeneratedProductCreation.getInt());
+	void getPetGroupShouldReturnPetGroup() {
+		PetGroup petGroup = PetGroup.values()[2];
+		when(random.nextInt(anyInt())).thenReturn(2);
+		
+		assertEquals(petGroup, autoGeneratedProductCreationStrategy.getPetGroup());
 	}
 	
 	@Test
-	void getPetGroup() {
-		when(strategy.getPetGroup()).thenReturn(PetGroup.CAT);
+	void getSeasonShouldReturnSeason() {
+		Season season = Season.values()[1];
+		when(random.nextInt(anyInt())).thenReturn(1);
 		
-		assertEquals(PetGroup.CAT, autoGeneratedProductCreation.getPetGroup());
+		assertEquals(season, autoGeneratedProductCreationStrategy.getSeason());
 	}
 	
 	@Test
-	void getSeason() {
-		when(strategy.getSeason()).thenReturn(Season.WINTER);
+	void getSizeShouldReturnSize() {
+		Size size = Size.values()[1];
+		when(random.nextInt(anyInt())).thenReturn(1);
 		
-		assertEquals(Season.WINTER, autoGeneratedProductCreation.getSeason());
+		assertEquals(size, autoGeneratedProductCreationStrategy.getSize());
 	}
 	
 	@Test
-	void getSize() {
-		when(strategy.getSize()).thenReturn(Size.S);
+	void getIntShouldReturnInteger() {
+		when(random.nextInt(anyInt())).thenReturn(10);
 		
-		assertEquals(Size.S, autoGeneratedProductCreation.getSize());
+		assertEquals(10, autoGeneratedProductCreationStrategy.getInt());
 	}
 	
 	@Test
-	void getStringName() {
-	//	when(strategy.getStringName()).thenReturn("Name");
+	void getStringNameShouldReturnString() {
+		when(random.nextInt(anyInt())).thenReturn(1).thenReturn(2).thenReturn(3).thenReturn(4).thenReturn(5)
+				.thenReturn(6).thenReturn(7).thenReturn(8).thenReturn(9).thenReturn(10);
+		StringBuilder stringBuilder = new StringBuilder();
+		for (int i = 1; i < 11; i++) {
+			char letter = alphabet[i];
+			stringBuilder.append(letter);
+		}
+		String result = stringBuilder.toString();
 		
-		assertEquals("Name", autoGeneratedProductCreation.getStringName());
+		assertEquals(result, autoGeneratedProductCreationStrategy.getStringName());
 	}
 	
 	@Test
-	void getStringBrandName() {
-	//	when(autoGeneratedProductCreation.getStringBrandName()).thenReturn("Brand");
+	void getStringBrandNameShouldReturnString() {
+		when(random.nextInt(anyInt())).thenReturn(1).thenReturn(2).thenReturn(3).thenReturn(4).thenReturn(5)
+				.thenReturn(6).thenReturn(7).thenReturn(8).thenReturn(9).thenReturn(10);
+		StringBuilder stringBuilder = new StringBuilder();
+		for (int i = 1; i < 11; i++) {
+			char letter = alphabet[i];
+			stringBuilder.append(letter);
+		}
```

Comment:
Can be moved to private methods for better readability.Also it will allow us to get rid of code duplication

Reply:
done